### DISCUSSION TITLE
Client - Server Shared Code Clarification

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#17226] [Help]
This might be more of a webpack question, but if we have a wrapper around the Fetch and it is used in both getStaticProps/getStaticPaths as well as in our client code, does this function get pulled into the client JS file and still stay in the server side files?  Is this done by webpack?

Just trying to figure out if we have a logger for client and a logger for server (that has a 'fs' dependency), if I should create a Decorator for the Fetch function (one for client logger and one for server logger) and then use those individually.  In order to avoid the 'fs' errors on the client.

**Comment by jamesmosier** [comment]
If it is use on the server (`getStaticProps`/`getStaticPaths`) then it will be bundled in the server code and if you also use it on the client it'll be present in the client bundle. So it sounds like having a separate logger for client vs server would be your best option to avoid issues. And yes, webpack handles that. You can also use this tool to determine what is eliminated on the client: https://next-code-elimination.now.sh/
